test/blackbox/test_case_channel_blacklist_01/node_sim_peer_01.c

```diff
@@ -0,0 +1,120 @@
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <pthread.h>
+#include <assert.h>
+#include <signal.h>
+#include "../common/common_handlers.h"
+#include "../common/test_step.h"
+#include "../common/network_namespace_framework.h"
+#include "../../utils.h"
+
+#define CHANNEL_PORT 1234
+
+bool test_channel_blacklist_disonnection_peer_01_running;
+bool test_case_signal_peer_restart_01;
+
+static void channel_receive_cb(meshlink_handle_t *mesh, meshlink_channel_t *channel, const void *dat, size_t len);
+static bool channel_accept(meshlink_handle_t *mesh, meshlink_channel_t *channel, uint16_t port, const void *dat, size_t len);
+
+static int client_id = -1;
+
+static bool channel_accept(meshlink_handle_t *mesh, meshlink_channel_t *channel, uint16_t port, const void *dat, size_t len) {
+	(void)dat;
+	(void)len;
+
+	assert(port == CHANNEL_PORT);
+
+	if(!strcmp(channel->node->name, "nut")) {
+		meshlink_set_channel_receive_cb(mesh, channel, channel_receive_cb);
+		//mesh->priv = channel;
+
+		return true;
+	}
+
+	return false;
+}
+
+/* channel receive callback */
+static void channel_receive_cb(meshlink_handle_t *mesh, meshlink_channel_t *channel, const void *dat, size_t len) {
+	(void)mesh;
+	(void)channel;
+	(void)dat;
+	(void)len;
+
+	if(len == 0) {
+		fprintf(stderr, "Channel closure\n");
+	}
+
+	if(!strcmp(channel->node->name, "nut")) {
+		if(!memcmp(dat, "test", 5)) {
+			assert(meshlink_channel_send(mesh, channel, "reply", 5) >= 0);
+		}
+	}
+
+	return;
+}
+
+static void log_message(meshlink_handle_t *mesh, meshlink_log_level_t level, const char *text) {
+	(void)mesh;
+
+	//fprintf(stderr, "\x1b[34m peer:\x1b[0m %s\n", text);
+}
+
+void *test_channel_blacklist_disonnection_peer_01(void *arg) {
+	struct timeval main_loop_wait = { 2, 0 };
+	mesh_arg_t *mesh_arg = (mesh_arg_t *)arg;
+	test_channel_blacklist_disonnection_peer_01_running = true;
+
+	// Run relay node instance
+
+	meshlink_handle_t *mesh;
+	mesh = meshlink_open(mesh_arg->node_name , mesh_arg->confbase, mesh_arg->app_name, mesh_arg->dev_class);
+	assert(mesh);
+	meshlink_set_log_cb(mesh, MESHLINK_DEBUG, log_message);
+	meshlink_set_channel_accept_cb(mesh, channel_accept);
+
+	// Join relay node and if fails to join then try few more attempts
+
+	if(mesh_arg->join_invitation) {
+    assert(meshlink_join(mesh, mesh_arg->join_invitation));
+	}
+
+	assert(meshlink_start(mesh));
+
+	// All test steps executed - wait for signals to stop/start or close the mesh
+
+	while(test_channel_blacklist_disonnection_peer_01_running) {
+		select(1, NULL, NULL, NULL, &main_loop_wait);
+
+		if(test_case_signal_peer_restart_01) {
+			meshlink_stop(mesh);
+			sleep(2);
```

Comment:
Is this call to `sleep(2)` really necessary? If so, that points to a bug in MeshLink, because `meshlink_stop()` should really only return when it's completely shut down.